wd_helper.au3

```diff
@@ -2906,6 +2906,71 @@ Func _WD_JsonActionKey($sType, $sKey, $iSuffix = Default)
 	Return SetError(__WD_Error($sFuncName, 0, $sJSON), 0, $sJSON)
 EndFunc   ;==>_WD_JsonActionKey
 
+
+; #FUNCTION# ====================================================================================================================
+; Name ..........: _WD_GetFreePort
+; Description ...:  Locate and return an available TCP port within a defined range
+; Syntax ........: _WD_GetFreePort($iMinPort,  $iMaxPort)
+; Parameters ....: $iMinPort            - Starting port number
+;                  $iMaxPort            - Ending port number
+; Return values .: Success - Available TCP port number
+;                  Failure - 0 
+; Author ........: Danp2
+; Modified ......:
+; Remarks .......:
+; Related .......:
+; Link ..........:
+; Example .......: No
+; ===============================================================================================================================
+Func _WD_GetFreePort($iMinPort, $iMaxPort)
+	Local $aPorts = __WinAPI_GetTcpTable()
+
+	If Not @error Then
+		For $iPort = $iMinPort To $iMaxPort
+			_ArraySearch($aPorts, $iPort, Default, Default, Default, Default, Default, 3)
+			If @error = 6 Then Return $iPort
+		Next
+	Endif
+
+	Return 0
```

Comment:
What about reporting `@error` ?

Reply:
I'll update to use __WD_Error like other functions.